### PR TITLE
Add CORS validation command line option.

### PR DESCRIPTION
**Pull request by bgravenorst**
## Pull Request Description

Added the `rest-api-cors-origins` command line option.

### CODE REVIEW

**Review by NicolasMassart — DISMISSED** [review]

File: docs/Reference/CLI/CLI-Syntax.md [hunk 1]
```diff
@@ -1074,6 +1074,46 @@ File containing the node's private key.
...
+
+Listed domains can access the node using HTTP REST API calls. If your client interacts with Teku
+using a browser app (such as a block explorer), add the client domain to the list.
+
+The default value is "none." If you don't list any domains, browser apps can't interact with your
+Teku node.
+
+!!! tip
+
+    For testing and development purposes, use `all` or `*` to accept requests from any domain.
```
Comment (abdelhamidbakhta) [inline docs/Reference/CLI/CLI-Syntax.md:1114]: `*` is supported but `all` is not supported for the moment. Can you ask if we should ? If yes i will add suport to `all` value as well.

**Review by abdelhamidbakhta — CHANGES_REQUESTED** [review]
All is not supported in the current implementation. If we don't want to support it then we should remove it from the documentation. If we do want to support it then we should update the implementation accordingly

File: docs/Reference/CLI/CLI-Syntax.md (hunk 1 above)
Comment (NicolasMassart) [reply docs/Reference/CLI/CLI-Syntax.md:1114]: Ideally yes, having `all` supported would be nice, but I don't think it's a priority as anyway the `all` or `*` setting is only recommended for dev and not prod.
I think updating this PR is going to be more pragmatic ;)

**Review by MadelineMurray — APPROVED** [review]

**Review by abdelhamidbakhta — APPROVED** [review]
Looks good to me.

**Review by NicolasMassart — APPROVED** [review]

**Pull request merged**